src/Data/List/Sort/InsertionSort.agda

```diff
@@ -0,0 +1,230 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- An implementation of insertion sort and its properties.
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Relation.Binary.Bundles using (DecTotalOrder)
+
+module Data.List.Sort.InsertionSort
+  {a ℓ₁ ℓ₂}
+  (O : DecTotalOrder a ℓ₁ ℓ₂)
+  where
+
+open import Data.Bool.Base using (true; false ; if_then_else_)
+open import Data.Empty using (⊥-elim)
+open import Function using (id)
+
+open import Data.List.Base using (List; []; _∷_)
+open import Data.List.Relation.Unary.Linked using ([]; [-] ; _∷_)
+open import Data.List.Relation.Binary.Pointwise
+  using (Pointwise ; [] ; _∷_ ; decidable ; setoid)
+
+open import Relation.Binary.Bundles using (Setoid)
+open import Relation.Binary.Definitions using (Decidable)
+open import Relation.Binary.PropositionalEquality
+  using (inspect ; [_])
+open import Relation.Binary.Properties.DecTotalOrder O
+  using (≰⇒≥ ; ≥-antisym)
+open import Relation.Nullary.Decidable.Core
+  using (does ; yes ; no ; map′)
+
+open DecTotalOrder O
+  renaming (Carrier to A ; refl to reflA ; trans to transA)
+  using (totalOrder ; _≤?_ ; _≤_ ;
+  module Eq ; _≈_ ; ≤-respʳ-≈ ; ≤-respˡ-≈ ; antisym)
+  
+open import Data.List.Relation.Binary.Equality.Setoid Eq.setoid
+  using (_≋_)
+open import Data.List.Relation.Binary.Permutation.Setoid Eq.setoid
+open import Data.List.Relation.Unary.Sorted.TotalOrder totalOrder
+  using (Sorted)
+open import Data.List.Sort.Base totalOrder
+open import Relation.Binary.Reasoning.Setoid (setoid Eq.setoid)
+  as SetoidReasoning using ()
```

Comment:
Why not simply
```suggestion
import Relation.Binary.Reasoning.Setoid (setoid Eq.setoid)
  as SetoidReasoning
```
Although the proliferation of `setoid`s makes it hard to see that this is declaring a reasoning module for `_≋_`... so maybe could be improved by renaming to
```suggestion
import Relation.Binary.Reasoning.Setoid (setoid Eq.setoid)
  as ≋-Reasoning
```